src/array.ts

```diff
@@ -230,29 +230,51 @@ export const select = <T, K>(
   }, [] as K[])
 }
 
+type GetterFunc<T extends number | object> = (item: T) => number
+
 /**
  * Max gets the greatest value from a list
  *
- * Ex. max([{ num: 1 }, { num: 2 }], x => x.num) == 2
+ * @example
+ * max([ 2, 3, 5]) == 5
+ * max([{ num: 1 }, { num: 2 }], x => x.num) == 2
```

Comment:
small nit: just to be clear to users because showing 2 could be confusing
```suggestion
 * max([{ num: 1 }, { num: 2 }], x => x.num) == { num: 2 }
```